packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -409,51 +408,20 @@ export class GraphQLApi extends GraphqlApiBase {
   constructor(scope: Construct, id: string, props: GraphQLApiProps) {
     super(scope, id);
 
-    this.validateAuthorizationProps(props);
-    const defaultAuthorizationType =
-      props.authorizationConfig?.defaultAuthorization?.authorizationType ??
-      AuthorizationType.API_KEY;
-
-    let apiLogsRole;
-    if (props.logConfig) {
-      apiLogsRole = new Role(this, 'ApiLogsRole', {
-        assumedBy: new ServicePrincipal('appsync'),
-      });
-      apiLogsRole.addManagedPolicy(
-        ManagedPolicy.fromAwsManagedPolicyName(
-          'service-role/AWSAppSyncPushToCloudWatchLogs',
-        ),
-      );
-    }
+    const defaultMode = props.authorizationConfig?.defaultAuthorization ??
+      { authorizationType: AuthorizationType.API_KEY };
+    const additionalModes = props.authorizationConfig?.additionalAuthorizationModes ?? [];
+    const modes = [defaultMode, ...additionalModes];
+
+    this.validateAuthorizationProps(modes);
```

Comment:
can this just be passed inline or is `modes` somewhere else?

Reply:
yeah its used elsewhere hence the variable

https://github.com/aws/aws-cdk/blob/d93969f0b540557cd943de5ae7ec72e1cd31130f/packages/%40aws-cdk/aws-appsync/lib/graphqlapi.ts#L435-L442